### DISCUSSION TITLE
New icon for profile social links

### DISCUSSION THREAD

**Discussion by loidnoir** [community/community#46378] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

I think it would be a great idea to add Discord icon support for profile social links, because it is common in GitHub to find user who mentioned his discord somewhere in his profile

**Comment by RONIE70** [comment]
Hello!!
Buenas prácticas para bots de Discord.
Este repositorio es básicamente un fork de [discord-bot-best-practices](https://github.com/meew0/discord-bot-best-practices/) pero traducido al Español y actualizado para seguir la [nueva política de privacidad para desarrolladores](https://discord.com/developers/docs/policy).

Importante: Estas pautas están destinadas a bots en servidores públicos. Si su bot es privado seguramente no le sea aplicable.

📏 Pautas
Los comandos deben ser llamados explícitamente. Los bots no deberían responder o ser llamados en un chat normal. Debe usar un prefijo para los comandos o solo responder cuando su bot es @mencionado.

Utilice prefijos únicos. Los prefijos de un solo carácter como !, $ y . son comunes para ejecutar comandos y dan lugar a coincidencias y problemas con otros bots. Si opta por usar un prefijo para su bot, considere usar palabras (estrella) o caracteres Unicode únicos (¨). Además, debes evitar usar # o @ como prefijos, ya que pueden usarse para mencionar un canal o un miembro. Idealmente, el prefijo de su bot debe ser configurable servidor por servidor, de modo que los propietarios del servidor puedan asegurarse de que cada bot tenga su propio prefijo único de su elección.

No seas codicioso. Limítese a una pequeña cantidad de prefijos para reducir el riesgo de coincidencias con otros.

No abuses de las menciones. Si responde directamente a un comando, no use una mención, pueden crear bucles de respuestas. Las menciones están bien si se ejecuta un comando de larga duración, los mensajes privados (MDs) son una buena alternativa.

Tenga un comando de info/botinfo. Debe proporcionar información sobre el bot, como qué librería está usando y la versión utilizada, comandos de ayuda (help) y, lo más importante, quién lo creó.

No responda con "comando no válido". Si un usuario usa un comando que no existe, déjelo fallar silenciosamente. No haga que responda con algo como "comando no válido". Aunque si el comando es correcto, pero los argumentos son incorrectos, entonces está bien responder con "argumentos no válidos". Si hay más de un bot en un servidor que comparte un prefijo, esto puede resultar en un uso muy desagradable. Si el prefijo de su bot es configurable, esta regla probablemente se pueda ignorar de manera segura.

Sea respetuoso con la API de Discord. Los bots que abusan y hacen mal uso de la API de Discord arruinan las cosas para todos. Respete los [Términos de servicio](https://discord.com/terms), las [Directivas de la comunidad](https://discord.com/guidelines) y [Política de privacidad para desarrolladores](https://discord.com/developers/docs/policy). Asegúrese de tener en cuenta la limitación de solicitudes y de integrarlo en el código de su bot, y sea inteligente al usar la API. Asegúrese de pedir ayuda si no está seguro de la forma correcta de implementar las cosas. Recuerde añadir "tiempos de espera" o "intervalos" para evitar abusar de la API de Discord.

Ignore sus propios mensajes y los de otros bots. Esto ayuda a prevenir bucles infinitos y posibles vulnerabilidades de seguridad. Usar un espacio "invisible" como \u200B y \u180E al comienzo de cada mensaje también evita que su bot active los comandos de otros bots. La API de Discord también le dice si un usuario es un bot (propiedad bot en el objeto User, [referencia](https://discord.com/developers/docs/resources/user#user-object)).

Mantenga las funciones [NSFW](https://support.discord.com/hc/es/articles/115000084051) solo en canales [NSFW](https://support.discord.com/hc/es/articles/115000084051) Todos los comandos/funciones NSFW solo deberían funcionar en canales marcados como NSFW.

Use las menciones al bot para ayudar a los usuarios. Permita una mención como prefijo ("@Bot ayuda"). O agrege una forma de encontrar el prefijo del bot con solo una mención ("@Bot" o "@Bot prefijo" ) ayudará a los usuarios que son nuevos en su bot a comenzar. (Asegúrate de que sea cual sea el mensaje, se encuentra fácilmente. Una excelente manera de hacerlo es incluyéndolo en presencia de tu bot). Además, una mención es el prefijo más exclusivo de todos.

Evite las menciones. Siempre que su bot responda a algo, asegúrese de que evite las menciones (especialmente @everyone y @here). Si bien es posible que el usuario no tenga permiso para mencionar a @everyone, su bot podría añadirlas inapropiadamente como por ejemplo, en el comando decir. Algunas bibliotecas hacen esto automáticamente de manera predeterminada, pero otras requerirán que configure las menciones. Consulta la sección de [Menciones permitidas](https://discord.com/developers/docs/resources/channel#allowed-mentions-object) de Discord.

Sea cuidadoso con los datos. Asegúrese de ser cuidadoso con los datos que recibe de la API de Discord. De ninguna manera intente dirigirse a los usuarios para mostrar publicidad, vender datos de los usuarios, revelar datos sensibles de los usuarios sin su consentimiento, alentar o promover actividades ilegales. Y un largo etcétera que [puede consultar](https://discord.com/developers/docs/policy).

**Comment by chris-c-thomas** [comment]
I do agree that more icons would be nice in general. What link would you share on Discord? Most user links are by ID such as https://discord.com/users/396054977291878400 - not a nice looking link which stinks on their part.

**Comment by Gakk** [comment]
I would like to see an icon for profile link to StackOverflow:

<img src="https://user-images.githubusercontent.com/5013909/218193468-d66d3927-afcc-4570-bf92-052bfd8518a7.png" width="64" />

With a link to profile, like: https://stackoverflow.com/users/146921/frode-evensen

**Comment by Faiqlatheef** [comment]
make it like below in your readme
<p align="center">
<a href="linkedin link" target="blank"><img align="center" src="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/linked-in-alt.svg" alt="ashutosh mishra" height="30" width="40" /></a>
<a href="instagram link" target="blank"><img align="center" src="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/instagram.svg" alt="simplified_learner" height="30" width="40" /></a>
<a href="twitter link" target="blank"><img align="center" src="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/twitter.svg" alt="simplified learner" height="30" width="40" /></a>
<a href="facebook link" target="blank"><img align="center" src="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/facebook.svg" alt="simplified learner" height="30" width="40" /></a>
</p>

**Reply by Faiqlatheef** [reply]
![image](https://user-images.githubusercontent.com/75663412/219350786-7b1d59dd-bba7-4e1e-a6ab-8e0eb450b093.png)

**Reply by Gakk** [reply]
I already have profile links in my personal readme just as you shows 🙂

This discussion is about GitHub's [new feature](https://github.blog/changelog/2023-02-02-add-more-social-links-to-your-user-profile/) for profile social links released a few weeks ago, and how it can be improved 😎
